### PR TITLE
REF: Categorical.is_dtype_equal -> categories_match_up_to_permutation

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

There are too many notions of CategoricalDtype equality.  This gives a clearer name to one of them.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -2133,7 +2134,7 @@ def _concat_same_type(self, to_concat):
 
     # ------------------------------------------------------------------
 
-    def is_dtype_equal(self, other):
+    def categories_match_up_to_permutation(self, other: "Categorical") -> bool:
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:2137]: umm, why this name? what is wrong with the existing one?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:2137]: > what is wrong with the existing one?

`is_dtype_equal(self, other)` means something different from `self.is_dtype_equal(other)` which i find confusing

>  why this name? 

it accurately describes what is being checked

Comment (jreback) [reply pandas/core/arrays/categorical.py:2129]: i don't have a problem with this as a private method, and its ok to deprecate the .is_dtype_equal but we shouldn't offer this as a replacement.

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py]: privatized+green

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -2138,8 +2139,17 @@ def is_dtype_equal(self, other):
         -------
         bool
         """
+        return hash(self.dtype) == hash(other.dtype)
+
+    def is_dtype_equal(self, other) -> bool:
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:2144]: we should deprecate any is_dtype_equal *methods* on any other Array/Index (not sure if we have any)

**Review by jreback — APPROVED** [review]

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:2144]: There's just a CategoricalDtype.is_dtype_equal left.  ill be happy to see that go

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:2144]: woops, no, got rid of that a while ago

**Pull request merged**